src/AutoRest.CSharp/MgmtTest/Generation/Mock/MgmtMockTestBaseWriter.cs

```diff
@@ -211,27 +284,85 @@ protected void WriteTestOperation(CodeWriterDeclaration declaration, MockTestCas
+                    {
+                        _writer.UseNamespace("System.Text.Json");
+                        _writer.Line($"Console.WriteLine($\"ResourceId: {{_.Data.Id}}\");");
+                        _writer.Line($"Console.WriteLine($\"Full Resource content: {{JsonSerializer.Serialize(_.Data)}}\");");
+                    }
+                }
             }
             else
             {
-                _writer.Append($"await ");
+                CodeWriterDeclaration operation = new CodeWriterDeclaration("operation");
+                if (generatingSample)
+                {
+                    _writer.Append($"var {operation:D} = ");
+                }
+                else
+                {
+                    _writer.Append($"await ");
+                }
                 WriteTestMethodInvocation(declaration, testCase);
                 _writer.LineRaw(";");
+                _writer.Line();
+                if (generatingSample)
+                {
+                    if (testCase.ClientOperation.ReturnType.Name == "ArmOperation")
+                    {
+                        _writer.UseNamespace("System.Text.Json");
+                        if (testCase.ClientOperation.ReturnType.Arguments.Length > 0 && testCase.ClientOperation.ReturnType.Arguments[0].Implementation is Resource)
+                        {
+                            _writer.Line($"Console.WriteLine($\"Succeed on ResourceId: {{{operation}.Value.Data.Id}}\");");
+                            _writer.Line($"Console.WriteLine($\"Full Resource content: {{JsonSerializer.Serialize({operation}.Value.Data)}}\");");
+                        }
+                        else
+                        {
+                            _writer.Line($"Console.WriteLine($\"Response: {{JsonSerializer.Serialize({operation}.Value)}}\");");
+                        }
+                    }
+                    else if (testCase.ClientOperation.ReturnType.Name == "Response" && testCase.ClientOperation.ReturnType.Arguments.Length > 0)
+                    {
+                        if (testCase.ClientOperation.ReturnType.Arguments[0].IsFrameworkType)
+                        {
+                            _writer.Line($"Console.WriteLine($\"Succeed with return: {{{operation}.Value}}\");");
+                        }
+                        else if (testCase.ClientOperation.ReturnType.Arguments[0].Implementation is Resource)
+                        {
+                            _writer.UseNamespace("System.Text.Json");
+                            _writer.Line($"Console.WriteLine($\"Succeed on ResourceId: {{{operation}.Value.Data.Id}}\");");
+                            _writer.Line($"Console.WriteLine($\"Full Resource content: {{JsonSerializer.Serialize({operation}.Value.Data)}}\");");
+                        }
+                        else
```

Comment:
Once again we should not have this complicated calculation in the writer, and if this piece of code is specific to sample generation, we should not put it in the base writer.